docs/mvvm/Ioc.md

```diff
@@ -0,0 +1,155 @@
+
+```csharp
+public class FileLogger : IFileLogger
+{
+    private readonly IFilesService FileService;
+    private readonly IConsoleService ConsoleService;
+
+    public FileLogger(
+        IFilesService fileService,
+        IConsoleService consoleService)
+    {
+        FileService = fileService;
+        ConsoleService = consoleService;
+    }
+
+    // Methods for the IFileLogger interface here...
+}
+```
+
+Here we have a `FileLogger` type implementing the `IFileLogger` interface, and requiring `IFilesService` and `IConsoleService` instances. Constructor injection means the DI service provider will automatically gather all the necessary services, like so:
+
+```csharp
+/// <summary>
+/// Configures the services for the application.
+/// </summary>
+private static IServiceProvider ConfigureServices()
+{
+    var services = new ServiceCollection();
+
+    services.AddSingleton<IFilesService, FilesService>();
+    services.AddSingleton<IConsoleService, ConsoleService>();
+    services.AddSingleton<IFileLogger, FileLogger>();
+
+    return services.BuildServiceProvider();
+}
+
+// Retrieve a logger service with constructor injection
+IFileLogger fileLogger = App.Current.Services.GetService<IFileLogger>();
+```
+
+The DI service provider will automatically check whether all the necessary services are registered, then it will retrieve them and invoke the constructor for the registered `IFileLogger` concrete type, to get the instance to return.
+
+## What about viewmodels?
+
+A service provider has "service" in its name, but it can actually be used to resolve instances of any class, including viewmodels! The same concepts explained above still apply, including constructor injection. Imagine we had a `ContactsViewModel` type, using an `IContactsService` and an `IPhoneService` instance through its constructor. We could have a `ConfigureServices` method like this:
+
+```csharp
+/// <summary>
+/// Configures the services for the application.
+/// </summary>
+private static IServiceProvider ConfigureServices()
+{
+    var services = new ServiceCollection();
+
+    // Services
+    services.AddSingleton<IContactsService, ContactsService>();
+    services.AddSingleton<IPhoneService, PhoneService>();
+
+    // Viewmodels
+    services.AddTransient<ContactsViewModel>();
+
+    return services.BuildServiceProvider();
+}
+```
+
+And then in our `ContactsView`, we would assign the data context as follows:
+
+```csharp
+public ContactsView()
+{
+    this.InitializeComponent();
+    this.DataContext = App.Current.Services.GetService<ContactsViewModel>();
+}
+```
+
+## More docs
+
+For more info about `Microsoft.Extensions.DependencyInjection`, see [here](/aspnet/core/fundamentals/dependency-injection).
+
+## Examples
+
+You can find more examples in the [unit tests](https://github.com/Microsoft/WindowsCommunityToolkit/blob/rel/7.0.0/
+UnitTests/UnitTests.Shared/Mvvm).
```

Comment:
@kyaa-dost think you added a newline here.

@Sergio0694 these should all be ..github.com/windows-toolkit/.. and not Microsoft/ - mind updating quick?